### PR TITLE
feat: tariff zones in map

### PR DESCRIPTION
**Pull request by mortennordseth**
This implements functionality to draw tariff zone boundaries in the map. 

image

image

Closes https://github.com/AtB-AS/kundevendt/issues/15775

### CODE REVIEW

**Comment by jonasbrunvoll** [comment]
As far as I can tell it looks good, but I'm thinking maybe @adriansberg could take a look at this PR as well to be sure😊

File: src/components/map/use-map-tariff-zones.ts [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+const ZONE_BOUNDARY_LAYER_ID = 'zone-boundary-layer';
+const ZONE_NAMES_LAYER_ID = 'zone-names-layer';
+
+export const useMapTariffZones = async (
+  mapRef: React.MutableRefObject<mapboxgl.Map | undefined>,
+) => {
+  const { language } = useTranslation();
+  const [isZonesVisible, setIsZonesVisible] = useState(false);
+  const [isStyleLoaded, setIsStyleLoaded] = useState(false);
+  const { data } = useSWR('tariffZones', getTariffZones, {
```
Comment (adriansberg) [inline src/components/map/use-map-tariff-zones.ts:19]: Should we wrap the `useSWR` call in a helper function to make this function body tidier?

File: src/components/map/use-map-tariff-zones.ts [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    revalidateOnFocus: false,
+    revalidateOnReconnect: false,
+    revalidateIfStale: false,
+  });
+
+  const addTariffZonesToMap = useCallback(
+    (map: mapboxgl.Map) => {
+      if (!data || !isStyleLoaded) return;
+      if (!map.getSource('tariff-zones')) {
```
Comment (adriansberg) [inline src/components/map/use-map-tariff-zones.ts:28]:
```suggestion
      if (!map.getSource(TARIFF_ZONE_SOURCE_ID)) {
```

**Review by adriansberg — APPROVED** [review]
Looks good

**Pull request merged**